app/View/Feeds/add.ctp

```diff
@@ -1,296 +1,180 @@
+                'field' => 'source_format',
                 'label' => __('Source Format'),
-                'div' => 'input clear',
-                'options' => $feed_types,
+                'options' => $dropdownData['feedTypes'],
+                'type' => 'dropdown',
                 'class' => 'form-control span6'
-        ));
-    ?>
-        <div id="HeadersDiv">
-    <?php
-            echo $this->Form->input('headers', array(
+            ],
+            [
+                'field' => 'headers',
                 'label' => __('Any headers to be passed with requests (for example: Authorization)'),
-                'div' => 'clear',
                 'class' => 'input-xxlarge',
-                'type' => 'textarea',
-                'placeholder' => __('Line break separated list of headers in the "headername: value" format')
-            ));
-    ?>
-            <div>
-                <span id="basicAuthFormEnable" class="btn btn-inverse quick-popover" style="line-height:10px; padding: 4px 4px;"><?php echo __('Add Basic Auth');?></span>
-                <div id="basicAuthForm" class="quick-form" style="display:none;">
-                        <fieldset>
-                            <div class="input">
-                                <label for="BasicAuthUsername"><?php echo __('Username');?></label>
-                                <input class="form-control" type="text" id="BasicAuthUsername"><br />
-                            </div>
-                            <div class="input">
-                                <label for ="BasicAuthPassword"><?php echo __('Password');?></label>
-                                <input class="form-control" type="text" id="BasicAuthPassword"><br />
-                            </div>
-                        </fieldset>
-                        <span class="btn-inverse btn" onClick="add_basic_auth();" style="line-height:10px; padding: 4px 4px;"><?php echo __('Add basic auth header'); ?></span>
-                </div>
-            </div><br />
-        </div>
-        <div id="OrgcDiv" class="optionalField">
-        <?php
-            echo $this->Form->input('orgc_id', array(
-                    'label' => __('Creator organisation'),
-                    'div' => 'input clear',
-                    'options' => $orgs,
-                    'class' => 'form-control span6'
-            ));
-        ?>
-        </div>
-        <div id="TargetDiv" class="optionalField">
-    <?php
-        echo $this->Form->input('fixed_event', array(
+                'type' => 'headers',
+                'placeholder' => __('Line break separated list of headers in the "headername: value" format'),
+                'rows' => 4,
+            ],
+            [
+                'field' => 'orgc_id',
+                'label' => __('Creator organisation'),
+                'options' => $dropdownData['orgs'],
+                'type' => 'dropdown',
+                'div' => ['id' => 'OrgcDiv', 'style' => 'display:none'],
```

Comment:
```
'div' => ['id' => 'OrgcDiv', 'style' => 'display:none', 'class' => 'optionalField'],
```
Include class `optionalField` for all optional fields (`orgc_id`, `fixed_event`, ...)